docs/architecture/005-admin-api.md

```diff
@@ -0,0 +1,111 @@
+# 005: admin api
+
+Authors: @Macroz @okahilak @opqdonut
+
+# Terms
+
+Item: Catalogue item, resource, form, workflow, or license.
+
+# Background
+
+Different items have the enabling/disabling and archiving/unarchiving
+operations in common. Previously, these operations have been done
+via the API using update endpoint, e.g., /api/workflow/update, which
+updates both enabled and archived status at once.
+
+# Current status
+
+Lately, we have added editing the content of an item for certain types
+of items (namely, workflow, catalogue item, and, in some cases, form).
+Editing is done using the update endpoint, i.e., /api/workflow/update,
+in the case of workflows, and a new edit endpoint, /api/form/edit, in
+the case of forms and catalogue items.
+
+The current situation has two problems:
+
+- The API is not consistent across different types of items,
+
+- The terms update and edit are easily confused.
+
+# The decision
+
+For each type of item, let's have the following API endpoints.
+
+Each is prefixed by /api/[type], where type is one of the following:
+catalogue-item, resource, form, workflow, license.
+
+- GET /. As before, used to get all items.
+
+- GET /[id]. As before, used to get a single item by id.
+
+- POST /create. As before, used to create a new item.
+
+- PATCH /edit. Used to change content of an item.
+
+- PATCH /archive. Used to archive/unarchive an item.
+
+- PATCH /enable. Used to enable/disable an item.
+
+
+Additionally for forms:
+
+- GET /[id]/editable. As before, used to get the 'editable' status of an item.
+
+
+Additionally for licenses:
+
+- POST /add_attachment. As before, used to add an attachment file to a license.
+
+- POST /remove_attachment. As before, used to remove an attachment file from
+  a license.
+
```

Comment:
Not huge fan of the underscores in endpoint names, would use rather POST/PUT for adding and DELETE for removing.

Reply:
Done, I changed underscores to hyphens.